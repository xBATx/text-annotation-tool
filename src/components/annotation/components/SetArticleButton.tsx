import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { SetArticleModalOpened } from 'src/actions';

interface Props {
  openModal: () => void;
}

export const SetArticleButton = ({ openModal }: Props) => (
  <span
    style={{
      color: 'white',
      cursor: 'pointer',
      padding: 5,
      textDecoration: 'underline',
    }}
    onClick={openModal}
  >
    New article
  </span>
);

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    openModal: () => dispatch(SetArticleModalOpened(true)),
  };
};

export default connect(
  null,
  mapDispatchToProps,
)(SetArticleButton);
