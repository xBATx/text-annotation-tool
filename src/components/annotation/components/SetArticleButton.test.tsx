import React from 'react';
import renderer from 'react-test-renderer';
import { SetArticleButton } from './SetArticleButton';

it('renders without crashing', () => {
  const wrapper = renderer
    .create(<SetArticleButton openModal={jest.fn} />)
    .toJSON();

  expect(wrapper).toMatchSnapshot();
});
