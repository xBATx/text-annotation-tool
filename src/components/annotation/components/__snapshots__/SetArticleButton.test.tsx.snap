// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span
  onClick={[Function]}
  style={
    Object {
      "color": "white",
      "cursor": "pointer",
      "padding": 5,
      "textDecoration": "underline",
    }
  }
>
  New article
</span>
`;
